apps/studio/components/interfaces/Settings/Infrastructure/InfrastructureConfiguration/EnablePhysicalBackupsModal.tsx

```diff
@@ -0,0 +1,143 @@
+    { ref },
+    {
+      refetchInterval,
+      refetchOnWindowFocus: false,
+      onSuccess: (data) => {
+        if (data.is_physical_backups_enabled) {
+          setRefetchInterval(false)
+          const regionKey = AWS_REGIONS_VALUES[selectedRegion]
+
+          if (ref === undefined) return toast.error('Project ref is required')
+          if (!regionKey)
+            return toast.error('Unable to deploy replica: Unsupported region selected')
+          const primary = databases?.find((db) => db.identifier === ref)
+          setUpReplica({
+            projectRef: ref,
+            region: regionKey as Region,
+            size: primary?.size ?? 't4g.small',
+          })
+        }
+      },
+    }
+  )
+
+  const { mutate: enablePhysicalBackups, isLoading: isEnabling } = useEnablePhysicalBackupsMutation(
+    {
+      onSuccess: () => {
+        setEnabling(true)
+        setRefetchInterval(5000)
+      },
+    }
+  )
+
+  const { mutate: setUpReplica } = useReadReplicaSetUpMutation({
+    onSuccess: () => {
+      const region = AVAILABLE_REPLICA_REGIONS.find((r) => r.key === selectedRegion)?.name
+      toast.success(`Spinning up new replica in ${region ?? ' Unknown'}...`)
+      setOpen(false)
+    },
+    onError: (error) => {
+      setEnabling(false)
+      toast.error(`Failed to deploy relica: ${error.message}`)
+    },
+  })
+
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogTrigger asChild>
+        <Button type="primary">Deploy replica</Button>
+      </DialogTrigger>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Physical backups are required to deploy replicas</DialogTitle>
+          <DialogDescription>
+            They are used under the hood to spin up read replicas for your project
+          </DialogDescription>
+        </DialogHeader>
+        <DialogSectionSeparator />
+        <DialogSection className="flex flex-col gap-y-2">
+          {enabling ? (
+            <>
+              <div className="flex items-center gap-x-3">
+                <Loader2 className="animate-spin" size={16} />
+                <p className="text-sm">Enabling physical backups</p>
+              </div>
+              <Markdown
+                content={`Please do not close the browser until this is completed. This will take a few minutes and your read replica will be deployed thereafter.`}
+              />
+            </>
+          ) : (
+            <>
+              <p className="text-sm">
+                Enabling physical backups will take a few minutes, after which your replica will be
+                deployed once completed.
+              </p>
+              <Markdown
+                content={`Do note that you will no longer be able to download your [scheduled backups](/projects/${ref}/database/backups/scheduled) once physical backups have been enabled for your project.`}
+              />
+            </>
+          )}
+        </DialogSection>
+        {!enabling && (
```

Comment:
ah I see what you're doing. This might not be great for when things go wrong, just getting shoved back to the 1st state. 

It's certainly nicer having multiple modal states (maybe we should be adding layers of dialogs on top) but when it fails, might should handle that failure in the next UI state and then offer a chance to try again.

Reply:
^ ahh yeah agree RE having multiple modal states, just FYI im dropping this UI change for now, reverting back to the old one so will park this